src/main/java/net/sf/jabref/gui/FileDialog.java

```diff
@@ -200,21 +176,23 @@ public FileDialog updateWorkingDirPref() {
      * @return The path of the new file, or {@link Optional#empty()} if dialog is aborted
      */
     public Optional<Path> saveNewFile() {
-        fileChooser.setDialogType(JFileChooser.SAVE_DIALOG);
-        if (showDialogAndIsAccepted()) {
-            File file = fileChooser.getSelectedFile();
-
-            if (!extensions.isEmpty() && !fileChooser.accept(file)) {
-                return Optional.of(Paths.get(file.getPath() + extensions.iterator().next().getFirstExtensionWithDot()));
-            }
-
-            return Optional.of(file.toPath());
-        }
-        return Optional.empty();
+        configureFileChooser();
+        return runInJavaFXThread(() -> Optional.ofNullable(fileChooser.showSaveDialog(null)).map(File::toPath));
```

Comment:
Please document, why `null` is passed here. We wanted to avoid passing `null`, but it seems the API required it.